mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -500,6 +503,55 @@ private[spark] object BLAS extends Serializable with Logging {
     nativeBLAS.dgemv(tStrA, mA, nA, alpha, A.values, mA, x.values, 1, beta,
       y.values, 1)
   }
+ 
+  /**
+   * y := alpha * A * x + beta * y
+   * For `DenseMatrix` A and SparseVector x.
+   */
+  private def gemv(
+      alpha: Double,
+      A: DenseMatrix,
+      x: SparseVector,
+      beta: Double,
+      y: DenseVector): Unit =  {
+    val mA: Int = A.numRows
+    val nA: Int = A.numCols
+
+    val Avals = A.values
+    var colCounterForA = 0
+
+    var xIndices = x.indices
+    var xNnz = xIndices.size
```

Comment:
`xIndices.length` when you try to get the size of native scala Array. `size` will call `length` which is another jvm call.